sql/parse/util.go

```diff
@@ -169,6 +199,26 @@ func readIdent(ident *string) parseFunc {
 	}
 }
 
+func readQuotedIdent(ident *string) parseFunc {
+	return func(r *bufio.Reader) error {
+		var buf bytes.Buffer
+		if err := readLetter(r, &buf); err != nil {
```

Comment:
why is this necessary? Doesn't this force the first character to be a letter?

Reply:
I copied that from `readIdent`, but I didn't think about whether that is necessary. The name `readLetter` does imply that the first character must be a letter, but since the test cases which have identifiers starting with non-letters pass, this seems to not be the case. Perhaps it has another purpose? Regardless, I agree that `readLetter` isn't needed here. Will remove.

Reply:
Rather than removing completely, I replaced it with a call to `readValidQuotedIdentRune` so that the identifier cannot be empty.